pandas/core/groupby/grouper.py

```diff
@@ -212,13 +208,13 @@ def groups(self):
 
     def __repr__(self) -> str:
         attrs_list = (
-            "{name}={val!r}".format(name=attr_name, val=getattr(self, attr_name))
+            f"{attr_name}={getattr(self, attr_name)!r}"
```

Comment:
can you do this in two steps so as to not have the getattr inside the fstring

Reply:
Not really sure how to do this in two steps since attr_name is generated from a list comprehension. Do I just drop list comprehension usage in that case?

Reply:
```
attr_tups = ((name, getatttr(self, name)) for name in self._attributes)
attr_tups = (x for x in attr_tups if x[1] is not None)
attrs_list = (f"{attr_name}={attr_val!r}" of attr_name, attr_val in attr_tups)
```

Reply:
this usage seems ok to me